pkg/provisioner/ironic/ironic.go

```diff
@@ -420,65 +420,13 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged bool) (r
 			imageData = p.host.Spec.Image
 		}
 
-		checksum, checksumType, ok := imageData.GetChecksum()
-
+		_, _, ok := imageData.GetChecksum()
 		if ok {
-			p.log.Info("setting instance info",
-				"image_source", imageData.URL,
-				"image_os_hash_value", checksum,
-				"image_os_hash_algo", checksumType,
-			)
-
-			updates := nodes.UpdateOpts{
-				nodes.UpdateOperation{
-					Op:    nodes.AddOp,
-					Path:  "/instance_info/image_source",
-					Value: imageData.URL,
-				},
-				nodes.UpdateOperation{
-					Op:    nodes.AddOp,
-					Path:  "/instance_info/image_os_hash_value",
-					Value: checksum,
-				},
-				nodes.UpdateOperation{
-					Op:    nodes.AddOp,
-					Path:  "/instance_info/image_os_hash_algo",
-					Value: checksumType,
-				},
-				nodes.UpdateOperation{
-					Op:    nodes.ReplaceOp,
-					Path:  "/instance_uuid",
-					Value: string(p.host.ObjectMeta.UID),
-				},
```

Comment:
So previously we were setting the instance_uuid here, but now we're not.
I can't recall if that was *required* for adoption (I thought it was?) but it sure makes debugging simpler.

Reply:
I feel like  it was required for ironic to believe the node was in use.

Reply:
Ah thanks, I missed that it got lost here, will fix!

Reply:
That's already set in getUpdateOptsForNode - I wonder if we should just call that here - it sets a few more things but we probably want all-the instance_info set on adoption?

Reply:
The rest are only required for actual provisioning. It might not hurt to set them all on adoption, but it might need more work to check that we have all of the info required - currently we set this stuff if there is an image in either the status or the spec, but an image in the spec doesn't guarantee that provisioning has actually started. e.g. it's entirely possible to have an image in the spec but no hardware profile set.
Come to think of it, we probably should be checking that the provision state has at least reached Provisioning before putting the image stuff in there.

Reply:
I see, so since `instance_uuid` isn't dependent on the hardware profile, perhaps I should have moved that into `getImageUpdateOptsForNode()`

I see now that in the current code we only set `instance_uuid` when `imageData.GetChecksum()` returns `ok` so moving the `instance_uuid` to `getImageUpdateOptsForNode()` would match the current behavior

Reply:
reinstated `instance_uuid`